spec/unit/http/service/report_spec.rb

```diff
@@ -33,6 +33,16 @@
 
       subject.put_report('report', report, environment: environment)
     end
+
+    it 'includes headers set via the :http_extra_headers setting' do
+      stub_request(:put, uri).with(headers: {'Example-Header' => 'real-thing'})
+
+      #Puppet[:http_extra_headers] = 'Example-Header:real-thing,another:thing'
+      #Puppet[:http_extra_headers] = ['Example-Header:real-thing', 'another:thing']
+      Puppet[:http_extra_headers] = {'Example-Header' => 'real-thing', 'another' => 'thing'}
```

Comment:
@joshcooper I want this to work when we pass in a hash, but it does not. Is this worth my time to try to make it work, or should I just keep it as only accepting an array or string?

Reply:
it's not working because `HttpExtraHeadersSetting` extends `Puppet::Settings::ArraySetting`? Maybe try extending `Puppet::Settings::BaseSetting` directly? That said we typically set settings as string values, and relying on the setting to munge it to an array, numeric, etc.